src/main/java/org/testng/internal/TestMethodWorker.java

```diff
@@ -53,13 +55,22 @@ public TestMethodWorker(IInvoker invoker,
                           List<IClassListener> listeners)
   {
     m_invoker = invoker;
-    m_methodInstances = testMethods;
     m_suite = suite;
     m_parameters = parameters;
     m_groupMethods = groupMethods;
     m_classMethodMap = classMethodMap;
     m_testContext = testContext;
     m_listeners = listeners;
+    List<IMethodInstance> localMethodInstances = Lists.newArrayList(testMethods);
```

Comment:
In fact, what I mean is: don't try to sort method outside the graph.
BTW, I propose we add a comment which explains the fix is a workaround for the graph issue and we create a new issue for the graph one with a link to the current pull request.
WDYT?